plugin/completion.py

```diff
@@ -40,11 +41,100 @@
+            self.view.hide_popup()
+
+    def on_modified_async(self):
+        if self.view.is_popup_visible():
+            # hide the docs popup on each keystoke
+            self.view.hide_popup()
+
+
 class LspResolveDocsCommand(sublime_plugin.TextCommand):
+    def run(self, edit: sublime.Edit, item) -> None:
+        detail = item.get('detail')
+        documentation = self.normalized_documentation(item)
+
+        # don't show the detail in the cooperate AC popup,
+        # if it is alread shown in the AC details filed.
+        self.is_detail_shown = bool(detail)
+
+        content = self.get_content(documentation, detail)
+        self.show_popup(content)
+
+        if not detail or not documentation:
+            # To make sure that the details or documentation fields don't exist
+            # we resove the completion item.
+            # If those fields appear after the item is resolved
+            # update the popup.
+            self.do_resolve(item)
+
+    def show_popup(self, content):
+        mdpopups.show_popup(
+            self.view,
+            content,
+            flags=sublime.COOPERATE_WITH_AUTO_COMPLETE,
+            max_width=480,
+            max_height=333,
+            allow_code_wrap=True
+        )
+
+    def update_popup(self, content):
+        mdpopups.update_popup(
+            self.view,
+            content
+        )
+
+    def do_resolve(self, item: dict) -> None:
+        session = session_for_view(self.view, 'completionProvider', self.view.sel()[0].begin())
+        if not session:
+            return
 
-    def run(self, edit: sublime.Edit) -> None:
-        self.view.show_popup('<div style="padding: 10px;">Showing documentation is under construction</div>',
-                             sublime.COOPERATE_WITH_AUTO_COMPLETE)
+        client = client_from_session(session)
+        if not client:
+            return
+
+        completion_provider = session.get_capability('completionProvider')
+        has_resolve_provider = completion_provider and completion_provider.get('resolveProvider', False)
+        if has_resolve_provider:
+            client.send_request(Request.resolveCompletionItem(item), self.handle_resolve_response)
+
+    def handle_resolve_response(self, item):
+        documentation = self.normalized_documentation(item)
+        detail = item.get('detail')
+
+        content = self.get_content(documentation, detail)
+        self.show_popup(content)
+
+    def get_content(self, documentation, detail):
+        content = ""
+        if detail and not self.is_detail_shown:
+            content += """<div class='highlight' style='margin: 10px'>
+                <p>{}</p>
+            </div>""".format(detail)
+
+        if documentation:
+            content += """<div style='margin: 10px'>
+                <p>{}</p>
+            </div>""".format(documentation)
+
+        return content
+
+    def normalized_documentation(self, item: Dict[str, Any]) -> str:
```

Comment:
I think I used to call this function `normalized_documentation`, but I think the name `minihtml` suits better. It is a function that transform LSP-human-readable-text into ST-minihtml.

Reply:
Perhaps move this function to plugin/core/views.py as we're going to need it in more places (see https://github.com/sublimelsp/LSP/issues/991).

Reply:
Created a minihtml function that accepts a string or a MarkupContent as the first argument.